### PR TITLE
ANDROID: Improve shortcuts

### PR DESCRIPTION
**Pull request by lephilousophe**
This PR does two things.
The shortcut support added recently is extended up to Nougat MR1 (7.1) as I missed a portability trick for this version.
Then, dynamic shortcuts are now supported.
This means that every time a game is started, a shortcut, available by long pressing the ScummVM icon in the launcher, is created and the history contains the 4 last launched games.
This also has the benefit to make shortcut pinning on launcher easier: you just have to long press in the shortcut list and move the shortcut on the launcher.
This should allow users without a launcher supporting shortcut widgets to still be able to pin them.

### CODE REVIEW

File: backends/platform/android/org/scummvm/scummvm/ShortcutCreatorActivity.java [hunk 1]
```diff
@@ -55,6 +55,34 @@ public class ShortcutCreatorActivity extends Activity {
...
+			parsedIniMap = null;
+		} catch(IOException ignored) {
+			parsedIniMap = null;
+		}
+
+		if (parsedIniMap == null) {
+			return;
+		}
+
+		Game game = Game.loadGame(parsedIniMap, gameId);
```
Comment (antoniou79) [inline backends/platform/android/org/scummvm/scummvm/ShortcutCreatorActivity.java:72]: We should check for game being returned as "null" here.

**Comment by antoniou79** [comment]
The code looks good to me. I only spotted one possible issue with loadGame() returning null but the code not handling it. 

Sorry, I won't be able to build and test soon locally, but it's got my vote to merge and then test from the daily builds.

File: backends/platform/android/org/scummvm/scummvm/ShortcutCreatorActivity.java (hunk 1 above)
Comment (lephilousophe) [reply backends/platform/android/org/scummvm/scummvm/ShortcutCreatorActivity.java:72]: Fixed, thank you!

**Comment by bluegr** [comment]
@lephilousophe so this PR looks to be OK to merge now, right?

**Comment by lephilousophe** [comment]
Yes, it's good for me. Thanks!

**Pull request merged**